core/common/src/geometry/CoordinateReferenceSystem.ts

```diff
@@ -0,0 +1,457 @@
+ *    units (meter, US Survey Feet, ...) relative to some origin.
+ *  - The non projected CRS (also named geographic CRS by EPSG nomenclature) that does
+ *    not require a projection (projection method = None) and horizontal coordinates are expressed
+ *    as longitude and latitude (see [[Cartographic]])
+ *  Horizontal Coordinate Systems rely on a projection to flatten the surface of an ellipsoid (see [[GeodeticEllipsoid]]) which is
+ *  the mathematical model of the Earth surface.
+ *  Explanations in more details of the various concepts can be obtained from other sources including the page on the subject
+ *  on itwinJS.org.
+ *  A few details are still required to grasp the model. Geographic Coordinate Reference Systems rely on the concept of geodetic datums
+ *  (see [[GeodeticDatum]]) to convert latitude/longitude from one frame of reference to another. Such geodetic datum will bind the ellipsoid
+ *  and possibly define transformation steps required to convert from the currently used geodetic datum to the common datum WGS84
+ *  used for worldwide data (such as most popular imagery data sources). If there are no rules established to convert to WGS84
+ *  or if those rules are secret then the horizontal CRS can be datum-less and must make direct use of the ellipsoid
+ *  to define the earth surface.
+ *  For this purpose either the geodetic datum must be specified using either datumId or datum properties or both. If
+ *  both these values are undefined then it is possible to define the ellipsoid using properties ellipsoidId or ellipsoid or both.
+ *  These two pairs of properties are mutually exclusive with datum related properties having precedence. If a datum or datumId is set then
+ *  ellipsoid properties will automatically be undefined. If datumId or datum is defined then attempts to set ellipsoidId or ellipsoid
+ *  will fail silently, the values remaining undefined. The ellipsoidId will not be a repeat of the ellipsoidId property part of the
+ *  geodetic datum definition.
+ *  @alpha
+ */
+export class HorizontalCRS implements HorizontalCRSProps {
+  /** The identifier of the horizontal CRS as stored in the dictionary or the service database */
+  public id?: string;
+  /** Used only for user-defined definitions that will typically use a GUID as id. A display name for the CRS that allows
+   ** a human to understand the nature of the definition
+   */
+  public name?: string;
+  /** Description */
+  public description?: string;
+  /** The source of the CRS definition. */
+  public source?: string;
+  /** If true then indicates the definition is deprecated. It should then be used for backward compatibility only.
+   *  If false or undefined then the definition is not deprecated.
+   */
+  public deprecated?: boolean;
+  /** The EPSG code of the CRS. If undefined or zero then there is no EPSG code associated. */
+  public epsg?: number;
```

Comment:
Why not just use 0 to mean "no EPSG" instead of also having `undefined` mean the same thing?

Reply:
Modified comment. Both backend and GeoCoord service returned undefined when zero was stored in the library so zero value was not possible.